packages/circuit-ui/components/ComparisonTable/components/PlanTable/PlanTable.tsx

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright 2025, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use client';
+
+import {
+  forwardRef,
+  type HTMLAttributes,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { Plus } from '@sumup-oss/icons';
+
+import { utilClasses } from '../../../../styles/utility.js';
+import {
+  TableHeader,
+  type TableHeaderProps,
+} from '../TableHeader/TableHeader.js';
+import { FeatureRow, type FeatureRowProps } from '../TableRow/FeatureRow.js';
+import { generateFromIndex, getFirstEightRows } from '../../utils.js';
+import { Button } from '../../../Button/index.js';
+import { applyMultipleRefs } from '../../../../util/refs.js';
+import { throttle } from '../../../../util/helpers.js';
+import { Compact } from '../../../Compact/index.js';
+import { useMedia } from '../../../../hooks/useMedia/index.js';
+import { Body } from '../../../Body/index.js';
+import { clsx } from '../../../../styles/clsx.js';
+
+import classes from './PlanTable.module.css';
+
+export interface FeatureSection {
+  title: string;
+  features: FeatureRowProps[];
+}
+
+export interface PlanTableProps extends HTMLAttributes<HTMLTableElement> {
+  /**
+   * A descriptive caption for the table.
+   */
+  caption: string;
+  /**
+   * The label for the show all features button.
+   */
+  showAllFeaturesLabel: string;
+  /**
+   * A list of sections with features to display.
+   */
+  sections: FeatureSection[];
+  /**
+   * A list of the items to compare.
+   */
+  headers: TableHeaderProps[];
+  /**
+   * An array of the two positions of the currently selected plans.
+   * the first index is the first plan, the second index is the second plan.
+   */
+  activePlans: number[];
+}
+
+export const PlanTable = forwardRef<HTMLTableElement, PlanTableProps>(
+  (
+    { caption, showAllFeaturesLabel, headers, sections, activePlans, ...props },
+    ref,
+  ) => {
+    const tableRef = useRef<HTMLTableElement>(null);
+    const isMobile = useMedia('(max-width: 767px)');
+    const isTablet = useMedia('(max-width: 767px) and (min-width: 480px)');
+    const [isCollapsed, setIsCollapsed] = useState(
+      sections.reduce(
+        (totalRows, section) => totalRows + section.features.length + 1,
+        0,
+      ) > 8,
```

Comment:
Let's move the `8` to a constant and use it on line 118 so it's clear where the `9` comes from.